brave/brave-core
# Improvements for the IPFS diagnostics page brave://ipfs

Resolves https://github.com/brave/brave-browser/issues/13173
- Add Restart button (speeds up applying config changes)
- If not installed, note "IPFS not installed" and button to install and start; other items should be grayed out to indicate missing functionality
- Additionally, display the Learn more about IPFS and privacy link described above (placement/appearance TBD once we know whether this diagnostic page can be included with the rest of Brave settings UI)
- Added buttons to open Web UI
- If the user has this diagnostic page open in one tab with their node stopped, and then starts their node from a different tab, this tab doesn't update unless it's reloaded.

![image](https://user-images.githubusercontent.com/2965009/107808887-5c303400-6d7b-11eb-83da-570dd5cf49bb.png)

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues/13173) for my issue

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
- Open brave://ipfs and play with buttons, all buttons should work as expected
- Open 2 different windows with brave://ipfs page and try to stop/start node in one, the second one should update own status automatically
